runtime/databricks/automl_runtime/forecast/prophet/model.py

```diff
@@ -117,35 +136,45 @@ def predict_timeseries(self, horizon: int = None) -> pd.DataFrame:
         """
         return self._predict_impl(horizon)
 
-    def predict(self, context: mlflow.pyfunc.model.PythonModelContext, input: pd.DataFrame) -> pd.DataFrame:
+    def predict(self, context: mlflow.pyfunc.model.PythonModelContext, input_df: pd.DataFrame) -> pd.DataFrame:
         """
         Predict API from mlflow.pyfunc.PythonModel
         :param context: A :class:`~PythonModelContext` instance containing artifacts that the model
                         can use to perform inference.
-        :param input: Input dataframe
+        :param input_df: Input dataframe
         :return: A pd.DataFrame with the forecast components.
         """
-        return self._predict_impl()
+        self._validate_cols(input_df, [self._time_col])
+        input_df.rename(columns={self._time_col: "ds"}, inplace=True)
```

Comment:
[Discussion] Why do we want to modify the `input_df`? I don't think it's consistent with the API other problem types. Users typically do `y = model.predict(X)`, expect `X` to be unchanged and `y` to be a new Series or DataFrame. I think this is the behavior of prophet's own `predict` function as well [code](https://github.com/facebook/prophet/blob/ba9a5a2c6e2400206017a5ddfd71f5042da9f65b/python/prophet/forecaster.py#L1205) (note `df.copy()`). Shall we follow that convention?

Reply:
The main issue is that prophet requires the input column name be ds (timestamp) and y(target).

Reply:
Could we make a copy of input_df instead of directly modifying it? From the linked code by Ying, that's what prophet does.

Reply:
I understand we need to rename the input column to `ds`, but we do not need to specify target column name at inference time, right? What I have in mind is

```
def predict(self, input_df):
  ds_df = pd.DataFrame({"ds": input_df[self._time_col]))  # Make a copy
  predict_df = self.model().predict(ds_df)
  return predict_df["yhat"]
```

Note that we return a `pd.Series`, not a `pd.DataFrame`. This is consistent with classification/regression, where user can either do `y = model.predict(X)` or `X["result"] = model.predict(X)`. What do you think?

Reply:
It is fine with univariate forecast. But for multi-series forecasting, we have to return a dataframe with id columns. In this case, we need to return the predicted value with column name. Any better suggestion on this?

Reply:
I think we can also return a `pd.Series` in multi-series forecasting, where the index of the returned series maps to the index of input dataframe one-by-one. If the user wants to add result back to their input, they can always do `X["result"] = model.predict(X)`, but by default, they get a result without the input modified `y = model.predict(X)`.

We can also check with ml-api@databricks.com for suggestions. +@mengxr .